### PR TITLE
more macro guards with CAAM builds

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**

### CODE REVIEW

File: wolfcrypt/src/port/devcrypto/wc_devcrypto.c [hunk 1]
```diff
@@ -122,7 +122,7 @@ int wc_DevCryptoCreate(WC_CRYPTODEV* ctx, int type, byte* key, word32 keySz)
         case CRYPTO_SHA2_512_HMAC:
             ctx->sess.cipher = 0;
             ctx->sess.mac    = type;
-            ctx->sess.mackey    = (uint8_t*)key;
+            ctx->sess.mackey    = (u_int8_t*)key;
```
Comment (JacobBarthelmeh) [inline wolfcrypt/src/port/devcrypto/wc_devcrypto.c:125]: for older compiler versions this was switched to u_int8_t

File: wolfssl/wolfcrypt/aes.h [hunk 2]
```diff
@@ -209,6 +205,9 @@ struct Aes {
     word32 y0;
 #endif
 #endif /* HAVE_AESGCM */
+#ifdef WOLFSSL_CAAM
```
Comment (JacobBarthelmeh) [inline wolfssl/wolfcrypt/aes.h:208]: pulled out of AES-GCM macro guard, it could be used with AES-CBC or other AES builds

**Review by dgarske — APPROVED** [review]

File: wolfcrypt/src/port/devcrypto/wc_devcrypto.c (hunk 1 above)
Comment (ynezz) [reply wolfcrypt/src/port/devcrypto/wc_devcrypto.c:125]: BTW this [change broke compilation](https://github.com/wolfSSL/wolfssl/issues/5693#issuecomment-1277340109) on systems using `musl` libc as it needs explicit `#include <sys/types.h>` to have this type available.

Comment (JacobBarthelmeh) [reply wolfcrypt/src/port/devcrypto/wc_devcrypto.c:125]: Thanks @ynezz for the report and analysis, am getting back to this now to work on a resolution. Apologies for the delay and hassle that the change caused.

**Pull request merged**